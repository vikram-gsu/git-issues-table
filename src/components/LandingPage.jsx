import React, { Component } from "react";
import ProjectIssueStatus from "./ProjectIssueStatus";
const captured_response = require("../data/captured_response.json");

const formatDate = dateString => {
  const dateObj = new Date(dateString);
  const options = {
    year: "numeric",
    month: "long",
    day: "numeric",
    hour: "numeric",
    minute: "numeric",
    second: "numeric",
    hour12: false,
    timeZoneName: "short"
  };
  return Intl.DateTimeFormat("en-US", options).format(dateObj);
};

class LandingPage extends Component {
  state = {
    issues: null
  }
  componentDidMount() {
    fetch("https://api.github.com/repos/facebook/react/issues")
      .then(response => response.json())
      .then(issues =>
        this.setState({
          issues: captured_response.map(response => ({
            ...response,
            created_at: formatDate(response.created_at),
            updated_at: formatDate(response.updated_at)
          }))
        })
      );

    // this.setState({
    //   issues: captured_response.map(response => ({
    //     ...response,
    //     created_at: formatDate(response.created_at),
    //     updated_at: formatDate(response.updated_at)
    //   }))
    // });
  }
  render() {
    return (
      this.state.issues && <>
        <ProjectIssueStatus issues={this.state.issues} />
      </>
    );
  }
}

export default LandingPage;
