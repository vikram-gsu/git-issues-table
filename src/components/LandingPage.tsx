import React, { Component } from "react";
import { Dimmer, Loader, Segment } from "semantic-ui-react";
import formatDate from "../lib/formatDate";
import ProjectIssueStatus from "./ProjectIssueStatus";
import ErrorMessage from "./ErrorMessage";
import { fetchIssues } from "../api/issues";
// const captured_response = require("../data/captured_response.json");

export type label = {
  url: string;
  id: number;
  name: string;
  color: string;
};
export type GitlabIssue = {
  number: number;
  title: string;
  created_at: string;
  updated_at: string;
  labels: [label?];
  state: string;
};

interface LandingPageState {
  issues: Array<GitlabIssue>;
  loading: Boolean;
  error?: any;
}

class LandingPage extends Component<{}, LandingPageState> {
  state = {
    issues: [],
    loading: true,
    error: undefined
  };
  componentDidMount() {
    this.setState({ loading: true });

    fetchIssues().then(async ({ status, statusText, data }) => {
      if (status !== 200) {
        this.setState({
          error: new Error(`${status} - ${statusText}`)
        });
      } else {
        const issues = await data;
        this.setState({
          issues: issues.map(
            ({
              number,
              title,
              created_at,
              updated_at,
              labels,
              state
            }: GitlabIssue) => ({
              number,
              title,
              labels,
              state,
              created_at: formatDate(created_at),
              updated_at: formatDate(updated_at)
            })
          ),
          loading: false
        });
      }
    });

    // this.setState({
    //   issues: captured_response.map(response => ({
    //     ...response,
    //     created_at: formatDate(response.created_at),
    //     updated_at: formatDate(response.updated_at)
    //   }))
    // });
  }
  componentDidCatch(error: any, errorInfo: any) {
    console.log(error, errorInfo);
  }
  render() {
    const { loading, issues, error } = this.state;
    if (error) return <ErrorMessage errorObj={error} />;
    return !this.state.loading ? (
      <>
        <ProjectIssueStatus issues={issues} />
      </>
    ) : (
      <Dimmer.Dimmable as={Segment} dimmed={loading}>
        <Dimmer active={loading} inverted>
          <Loader active={loading}>Loading</Loader>
        </Dimmer>
      </Dimmer.Dimmable>
    );
  }
}

export default LandingPage;
