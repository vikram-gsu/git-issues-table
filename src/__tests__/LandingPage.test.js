import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import LandingPage from '../components/LandingPage';

describe('<LandingPage />', () => {
  it('renders landing page without crashing', () => {
    shallow(<LandingPage />)
  });

  it('renders landing page as expected', () => {
    const wrapper = shallow(<LandingPage />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });


});