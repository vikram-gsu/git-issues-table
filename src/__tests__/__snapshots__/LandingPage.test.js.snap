// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders landing page as expected 1`] = `
<DimmerDimmable
  as={[Function]}
  dimmed={true}
>
  <Dimmer
    active={true}
    inverted={true}
  >
    <Loader
      active={true}
    >
      Loading
    </Loader>
  </Dimmer>
</DimmerDimmable>
`;
